src/components/Modal/LoginModal.js

```diff
@@ -1,32 +1,34 @@
-import React, { useState } from "react";
+import React from "react";
 import { ImFacebook2 } from "react-icons/im";
 import { BsGoogle } from "react-icons/bs";
 import { useHistory } from "react-router-dom";
-
+import ReactDom from "react-dom";
 import "./LoginModal.css";
 import Home from "../Home/Home";
-import bazarApi from "../../features/api/bazarApi";
+import { useFormik } from "formik";
+import { useDispatch } from "react-redux";
+import { getUser } from "../../features/redux/userSlice";
 
 const LoginModal = () => {
+  const dispatch = useDispatch();
   const history = useHistory();
-  const [identifier, setIdentifier] = useState("");
-  const [password, setPassword] = useState("");
-
-  const loginHandler = async (e) => {
-    e.preventDefault();
 
-    let item = { identifier, password };
-    const res = await bazarApi.post("auth/local/", item);
-    localStorage.setItem("user-info", JSON.stringify(res.data));
+  const formik = useFormik({
+    initialValues: {
+      password: "",
+      identifier: "",
+    },
+    onSubmit: async (values) => {
+      let item = { identifier: values.identifier, password: values.password };
+      dispatch(getUser(item)).then((res) => {
+        if (!res.error) {
+          history.push("/user");
+        }
+      });
+    },
+  });
 
-    if (res.data.user) {
-      history.push("/user");
-    } else {
-      return;
-    }
-  };
-
-  return (
+  return ReactDom.createPortal(
     <div className="signup-modal">
       <Home />
```

Comment:
u are using portal incorrectly - portal just moves required component to any container so why do u add <Home /> here if the only thing u have to move is modal? :)